.github/workflows/release_github.yml

```diff
@@ -4,33 +4,79 @@ on:
   workflow_dispatch:
     inputs:
       version_level:
-        description: "Semantic version level increase."
+        description: Semantic version level increase
         required: true
         type: choice
+        default: patch
         options:
           - patch
           - minor
           - major
-
+          
 permissions:
   contents: write
-  pull-requests: write
   packages: read
+  statuses: read
+  checks: read
+  pull-requests: write
+  actions: read
+  repository-projects: read
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GH_PAGER: cat
 
 jobs:
+  checks:
+    name: Check requirements
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Fail if main branch was selected
+        if: ${{ github.ref_name == 'main' }}
+        run: |
+          echo "Cannot release from main branch, please select valid release branch."
+          exit 1
+
+      - name: Checkout repository
+        if: ${{ github.ref_name != 'develop' }}
+        uses: actions/checkout@v4
+
+      - name: Check if PR exists
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }}
```

Comment:
could you add a message here explaining why it fails, similar to what you did below?

Reply:
I think the error the `gh` command gives is clear enough. 

```
Run gh pr view test11
no pull requests found for branch "test11"
Error: Process completed with exit code 1.
```